frappe/public/js/frappe/form/controls/date.js

```diff
@@ -142,12 +142,84 @@ frappe.ui.form.ControlDate = class ControlDate extends frappe.ui.form.ControlDat
 			}
 		});
 	}
+
+	eval_expression(value, type) {
+		if (!value || !type) {
+			return value;
+		}
+		const parsed_components = value.match(/[\+\-]\s*\d*\.?\d+\s*[a-zA-Z]/g);
+
+		let system_fmt;
+		switch (type) {
+			case "date":
+				system_fmt = "YYYY-MM-DD";
+				value = frappe.datetime.user_to_str(value, false);
+				break;
+
+			case "time":
+				system_fmt = "HH:mm:ss";
+				value = frappe.datetime.user_to_str(value, true);
+				break;
+
+			case "datetime":
+				system_fmt = "YYYY-MM-DD HH:mm:ss";
+				value = frappe.datetime.user_to_str(value, false);
+				break;
+		}
```

Comment:
Just a question here for my understanding:

Where are these constants declared?

Reply:
`system_fmt` declared just above the switch, and `value` is a parameter of this function

Reply:
its not a constant variable

Reply:
Absolutely! I should have been more specific. :smile: 

I mean: where is the string value derived from?

Why is it `YYYY-MM-DD` and not `DD-MM-YYYY` for example?

If it's a variable, then it probably should not be set to a constant string, I think?

That's the reason I asked.

Reply:
`user_to_str` will return the value in system format, so I declare the format for each type. and the format is used for adding a datetime `value = moment(value, system_fmt).add(time_to_add).format(system_fmt);`.

Maybe, I can simply use 
`frappe.defaultDateFormat`
`frappe.defaultTimeFormat`
`frappe.defaultDatetimeFormat`

Reply:
@blaggacao did the above message clarify your question?

Reply:
I thought you wanted to use the `frappe.*` namespaced variables? :smile:

Reply:
Could you also check of a different format works, e.g. `11.10.2024+4d` -> `15.10.2024`

Reply:
Thinking of which, maybe you could use a locale-aware parsing lib / function?

I think it would be otherwise pretty hard to actually get this feature right, wdyt?

Reply:
`frappe.datetime.user_to_str` converts the user format into system format. so, we are only using the datetime in the system formats.

https://github.com/user-attachments/assets/e7449a8d-7bfb-42c5-8055-f3dbef268a7f

Reply:
Wonderful! Thanks for the proof!!! That was really helpful!